bitcoin/src/network/constants.rs

```diff
@@ -173,6 +173,44 @@ impl Network {
     }
 }
 
+#[cfg(feature = "serde")]
+pub mod as_core_arg {
+    //! Module for serialization/deserialization of network variants into/from Bitcoin Core values
+    #![allow(missing_docs)]
+    
+    use crate::Network;
+    use serde;
+
+    pub fn serialize<S>(network: &Network, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(network.to_core_arg())
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Network, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct NetworkVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for NetworkVisitor {
+            type Value = Network;
+    
+            fn visit_str<E: serde::de::Error>(self, s: &str) -> Result<Self::Value, E> {
+                Network::from_core_arg(s)
+                    .map_err(|_| E::invalid_value(serde::de::Unexpected::Str(s), &"bitcoin network encoded as a string"))
+            }
+    
+            fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result {
+                write!(formatter, "bitcoin network encoded as a string")
```

Comment:
Just realized, it might be nice to enumerate all possible networks in the message.

Reply:
Done in [4e9ae25](https://github.com/rust-bitcoin/rust-bitcoin/pull/1801/commits/4e9ae25626944e42ba0d54cb61de2e9918113e2f)

Reply:
Done in aa6a3530a626c17a875c0d961ece2ef8ab73d7fb